workflows: l4lb/verifier: fix skip-test-run job

jibi opened the pull request:
this job is currently failing as the sha parameter of the github-status
action is left unset, as we are skipping the "Retrieve pull request's
base and head" step.

This commit fixes this by setting the appropriate workflow sha variable
in the previous step.

aanm commented on .github/workflows/tests-datapath-verifier.yaml:
```diff
@@ -113,6 +113,13 @@ jobs:
+      run: |
+        curl ${{ github.event.issue.pull_request.url }} > pr.json
+        echo "base=$(jq -r '.base.sha' pr.json)" >> $GITHUB_OUTPUT
+        echo "head=$(jq -r '.head.sha' pr.json)" >> $GITHUB_OUTPUT
```
This should probably be 
```
      - name: Set up job variables
        id: vars
        run: |
          if [ ${{ github.event.issue.pull_request || github.event.pull_request }} ]; then
            PR_API_JSON=$(curl \
              -H "Accept: application/vnd.github.v3+json" \
              -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
              ${{ github.event.issue.pull_request.url || github.event.pull_request.url }})
            SHA=$(echo "$PR_API_JSON" | jq -r ".head.sha")
          else
            SHA=${{ github.sha }}
          fi
          echo "sha=${SHA}" >> $GITHUB_OUTPUT
```

jibi committed:
workflows: l4lb/verifier: fix skip-test-run job

this job is currently failing as the sha parameter of the github-status
action is left unset, as we are skipping the "Retrieve pull request's
base and head" step.

This commit fixes this by setting the appropriate workflow sha variable
in the previous step.

Fixes: #23847
Signed-off-by: Gilberto Bertin <jibi@cilium.io>

aanm reviewed (approved):
Thank you!
